cumulusci/tasks/bulkdata/tests/test_load.py

```diff
@@ -488,144 +488,76 @@ def test_get_statics_record_type_not_matched(self):
 
-    @mock.patch("cumulusci.tasks.bulkdata.load.aliased")
-    def test_query_db__joins_self_lookups(self, aliased):
-        task = _make_task(
-            LoadData, {"options": {"database_url": "sqlite://", "mapping": "test.yml"}}
-        )
-        model = mock.Mock()
-        task.models = {"accounts": model}
-        task.metadata = mock.Mock()
-        task.metadata.tables = {"accounts_sf_ids": mock.Mock()}
-        task.session = mock.Mock()
-
-        model.__table__ = mock.Mock()
-        model.__table__.primary_key.columns.keys.return_value = ["sf_id"]
-        columns = {"sf_id": mock.Mock(), "name": mock.Mock()}
-        model.__table__.columns = columns
-
-        mapping = MappingStep(
-            sf_object="Account",
-            table="accounts",
-            action=DataOperationType.UPDATE,
-            fields={"Id": "sf_id", "Name": "name"},
-            lookups={
-                "ParentId": MappingLookup(
-                    table="accounts", key_field="parent_id", name="ParentId"
-                )
-            },
-        )
-
-        task._query_db(mapping)
-
-        # Validate that the column set is accurate
-        task.session.query.assert_called_once_with(
-            model.sf_id,
-            model.__table__.columns["name"],
-            aliased.return_value.columns.sf_id,
-        )
-
-        # Validate that we asked for an outer join on the self-lookup
-        aliased.assert_called_once_with(task.metadata.tables["accounts_sf_ids"])
-        task.session.query.return_value.outerjoin.assert_called_once_with(
-            aliased.return_value, False
-        )
-
-    @mock.patch("cumulusci.tasks.bulkdata.load.aliased")
-    def test_query_db__person_accounts_enabled__account_mapping(self, aliased):
-        task = _make_task(
-            LoadData, {"options": {"database_url": "sqlite://", "mapping": "test.yml"}}
-        )
-        model = mock.Mock()
-        task.models = {"accounts": model}
-        task.metadata = mock.Mock()
-        task.metadata.tables = {"accounts_sf_ids": mock.Mock()}
-        task.session = mock.Mock()
-        task._can_load_person_accounts = mock.Mock(return_value=True)
-        task._filter_out_person_account_records = mock.Mock()
-
-        model.__table__ = mock.Mock()
-        model.__table__.primary_key.columns.keys.return_value = ["sf_id"]
-        columns = {
-            "sf_id": mock.Mock(),
-            "name": mock.Mock(),
-            "IsPersonAccount": mock.Mock(),
-        }
-        model.__table__.columns = columns
-
-        mapping = MappingStep(
-            sf_object="Account",
-            table="accounts",
-            action=DataOperationType.UPDATE,
-            fields={"Id": "sf_id", "Name": "name"},
-            lookups={
-                "ParentId": MappingLookup(
-                    table="accounts", key_field="parent_id", name="ParentId"
-                )
-            },
+    def test_query_db__joins_self_lookups(self):
```

Comment:
This testing strategy is nice because of the elision of a thousand mocks. Do you think there is potential fragility to changes in SQLAlchemy, though? I can't say that I know how stable their SQL marshalling is.

Reply:
I don't think it's a deal-breaker in any case, but wanted to think it through.

Reply:
Yes I'm a bit worried about that too but it seemed that:

a) I need to either make the code sensitive to changes in SQL Alchemy or changes in our code's implementation details.

b) if a test breaks it's easier to eyeball two SQL statements to see if they are logically equivalent than weird mocks.